src/features/counterfactual/ActivateAccount.tsx

```diff
@@ -0,0 +1,186 @@
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import useDeployGasLimit from '@/features/counterfactual/hooks/useDeployGasLimit'
+import { removeUndeployedSafe, selectUndeployedSafe } from '@/features/counterfactual/store/undeployedSafeSlice'
+import useChainId from '@/hooks/useChainId'
+import { useCurrentChain } from '@/hooks/useChains'
+import useGasPrice, { getTotalFee } from '@/hooks/useGasPrice'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { useWeb3 } from '@/hooks/wallets/web3'
+import { asError } from '@/services/exceptions/utils'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { hasFeature } from '@/utils/chains'
+import { formatVisualAmount } from '@/utils/formatters'
+import { Alert, Box, Button, CircularProgress, Divider, Grid, Typography } from '@mui/material'
+import type { DeploySafeProps } from '@safe-global/protocol-kit'
+import { FEATURES } from '@safe-global/safe-gateway-typescript-sdk'
+import React, { useContext, useState } from 'react'
+
+import TxLayout from '@/components/tx-flow/common/TxLayout'
+
+const ActivateAccountFlow = () => {
+  const [isSubmittable, setIsSubmittable] = useState<boolean>(true)
+  const [submitError, setSubmitError] = useState<Error | undefined>()
+
+  const chain = useCurrentChain()
+  const chainId = useChainId()
+  const { safeAddress } = useSafeInfo()
+  const provider = useWeb3()
+  const undeployedSafe = useAppSelector((state) => selectUndeployedSafe(state, chainId, safeAddress))
+  const { setTxFlow } = useContext(TxModalContext)
+  const dispatch = useAppDispatch()
+
+  const { gasLimit } = useDeployGasLimit()
+  const [gasPrice] = useGasPrice()
+
+  if (!undeployedSafe) return null
+
+  const isEIP1559 = chain && hasFeature(chain, FEATURES.EIP1559)
+  const maxFeePerGas = gasPrice?.maxFeePerGas
+  const maxPriorityFeePerGas = gasPrice?.maxPriorityFeePerGas
+
+  const totalFee =
+    gasLimit && maxFeePerGas
+      ? formatVisualAmount(getTotalFee(maxFeePerGas, maxPriorityFeePerGas, gasLimit), chain?.nativeCurrency.decimals)
+      : '> 0.001'
+
+  const createSafe = async () => {
+    if (!provider) return
+
+    setIsSubmittable(false)
+    setSubmitError(undefined)
+
+    try {
+      const options: DeploySafeProps['options'] = isEIP1559
+        ? {
+            maxFeePerGas: maxFeePerGas?.toString(),
+            maxPriorityFeePerGas: maxPriorityFeePerGas?.toString(),
+            gasLimit: gasLimit?.toString(),
+          }
+        : { gasPrice: maxFeePerGas?.toString(), gasLimit: gasLimit?.toString() }
+
+      await createNewSafe(provider, {
+        safeAccountConfig: undeployedSafe.safeAccountConfig,
+        saltNonce: undeployedSafe.safeDeploymentConfig?.saltNonce,
+        options,
+      })
+      dispatch(removeUndeployedSafe({ chainId, address: safeAddress }))
+    } catch (_err) {
+      const err = asError(_err)
+      setIsSubmittable(true)
+      setSubmitError(err)
+      return
+    }
+
+    setTxFlow(undefined)
+  }
```

Comment:
Please move some of this into a hook.

Reply:
It doesn't have to be reusable, just literally copy and paste everything into a use-function in the same or a separate file.

Reply:
I extracted the logic for getting the total formatted fee and moved the rest of the logic into a local hook.